src/viam/ur/module/ur_arm.cpp

```diff
@@ -525,6 +525,65 @@ URArm::KinematicsData URArm::get_kinematics(const ProtoStruct&) {
     return KinematicsDataSVA({temp_bytes.begin(), temp_bytes.end()});
 }
 
+// Unknown arm models should return an empty map. Arm models that do not have all expected parts in the map should return as much as they
+// have
+std::map<std::string, mesh> URArm::get_3d_models(const ProtoStruct&) {
+    const std::shared_lock rlock{config_mutex_};
+    check_configured_(rlock);
+
+    std::map<std::string, std::vector<std::string>> arm_name_to_model_parts;
+    arm_name_to_model_parts["ur5e"] = {"base_link", "ee_link", "forearm_link", "upper_arm_link", "wrist_1_link", "wrist_2_link"};
+
+    const std::string model_string = [&] {
+        if (model_ == model("ur3e")) {
+            return "ur3e";
+        } else if (model_ == model("ur5e")) {
+            return "ur5e";
+        } else if (model_ == model("ur7e")) {
+            return "ur7e";
+        } else if (model_ == model("ur20")) {
+            return "ur20";
+        }
+        return "";
+    }();
+
+    if (arm_name_to_model_parts.find(model_string) == arm_name_to_model_parts.end()) {
```

Comment:
If you just capture the resulting iterator, you don't need to do another lookup below on line 555.

```
const auto where = arm_name_to_model_parts.find(model_string);
if (where == arm_name_to_model_parts.end() {
    return {};
}
const auto& parts_to_load = *where;
```

This also avoids copying the result.